src/core/include/openvino/op/util/rnn_cell_base.hpp

```diff
@@ -97,6 +97,17 @@ class OPENVINO_API RNNCellBase : public Op {
     ///
     void validate_input_rank_dimension(const std::vector<PartialShape>& input);
 
+    ///
+    /// \brief      Validates static rank and dimension for provided input parameters.
+    ///             Additionally input_size dimension is checked for X and W inputs.
+    ///             Applies to LSTM, GRU and RNN Sequences.
+    ///
+    ///
+    /// \param[in]  input        Vector with RNNSequence-like op inputs in following order:
+    ///                          X, initial_hidden_state, sequence_lengths, W, R and B.
+    ///
+    static void validate_seq_input_rank_dimension(const std::vector<ngraph::PartialShape>& input);
```

Comment:
@jane-intel I've moved `validate_seq_input_rank_dimension` (not exposed it `Core` ealier) here from `util/reccurent_sequence`. I've made it `static` to make it usable for `op::v0::LSTMSequence` which doesn't inherrit from `RNNCellBase`. Should we move `validate_seq_input_rank_dimension` somewhere else or make `op::v0::LSTMSequence` child of `RNNCellBase` or just expose it in API in old location or a current version is ok? I don't have a strong opition here

Reply:
Or if we move shape inference of all sequence ops to `shape_inference` folder the problem with `validate_seq_input_rank_dimension` will resolve itself

Reply:
Here you added it as a public method which is backward-compatibility protected - meaning we will have no option to remove it in the future. Let's have it in the previous place but exposed as TRANSFORMATION_API -- is it possible though?
Because changing inhering for LSTMSequence is not considered safe. Code which uses dynamic pointer casting to RNNBaseCell would run for LSTMSequence while it didn't previously.

Reply:
ok, done